packages/web-app-files/src/components/SideBar/Shares/FileLinks.vue

```diff
@@ -375,12 +339,66 @@ export default defineComponent({
       return l1Direct ? -1 : 1
     },
 
+    checkPasswordEnforcedFor(link) {
+      const currentRole = this.availableRoleOptions.find(({ role }) => {
+        return parseInt(link.permissions) === role.bitmask(false)
+      })
+
+      const canRead = currentRole.role.hasPermission(SharePermissions.read)
+      const canCreate = currentRole.role.hasPermission(SharePermissions.create)
+      const canDelete = currentRole.role.hasPermission(SharePermissions.delete)
+
```

Comment:
What about the `update` permission for all the checks below?

Reply:
Good point, I went by the roles (instead of permissions) and since enforcement is an oc10-only capability as of now and there is no "Updater" role I'm aware of this was forgotten. Adding it now